src/components/MainPage.tsx

```diff
@@ -60,6 +77,16 @@ const MainPageComponent: React.FC = () => {
     setCurrentChatHistory(chatHistory)
   }
 
+  const openTabContent = async (path: string) => {
+    if (!path) return
+    if (isFilePath(path)) openFileAndOpenEditor(path)
```

Comment:
we should not have isFilePath as a string logic function. instead please modify this function to use this logic to determine whether to open a chat or a file:
chatHistoriesMetadata.find((chat) => chat.displayName === path)!.id

Reply:
and actually please move the logic of this function inside the tabs provider, it is easier to understand if the logic to decide whether to open a chat or a file is in the provider itself. 

(and also please don't use the isFilePath function as mentioned above)

Reply:
@joseplayero  This function is used for both tab provider and TitleBar, if we move this down to tab provider it would mean we need to implement the same in TitleBar again, creating a possible diverge. My two cents is to keep it in MainPage and pass it down to TabsProvider and TitleBar. WDYT? 

I agree on your first point of changing isFilePath

Reply:
Ah did not see that @yaodingyd ! 

in that case, could you just do the isFilePath change + fix the UI bugs in the above comment